### PR TITLE
Add deprecation notice Fixes #50

### LINKED ISSUE #50: setup.py test is going to be deprecated
from https://github.com/pytest-dev/pytest/issues/5534

this project also relies on setuptools options that are now considered 'smells':

```
    setup_requires=[
        'pytest-runner',
    ],
    tests_require=[
        'pytest',
    ],
```

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -19,6 +19,21 @@
...
+
+It is recommended that you:
+
+- Remove 'pytest-runner' from your 'setup_requires', preferably removing the `setup_requires` option.
+- Remove 'pytest' and any other testing requirements from 'tests_require', preferably removing the `setup_requires` option.
+- Use a requirement-test.txt file to hold test dependencies
+- Invoke tests with ``pytest``
+- Another good option is to configure a ``tox.ini`` next to your ``setup.cfg`` https://tox.readthedocs.io/en/latest/
+
 Usage
```
Comment (graingert) [inline README.rst]: Do you think it's worth removing this part of the docs, in case people scroll past the deprecation warning?

File: README.rst [hunk 2]
```diff
@@ -19,6 +19,21 @@
 Setup scripts can use pytest-runner to add setup.py test support for pytest
 runner.
 
+Deprecation Notice
+==================
+
+pytest-runner depends on deprecated features of setuptools and relys on features that break security
```
Comment (graingert) [inline README.rst]:
```suggestion
pytest-runner depends on deprecated features of setuptools and relies on features that break security
```

**Comment by graingert** [comment]
@jaraco I recommend archiving the project after merging this in, and making one more release to update PyPI long_description and rtfd:

https://help.github.com/en/articles/about-archiving-repositories

File: README.rst [hunk 3]
```diff
@@ -19,6 +19,21 @@
 Setup scripts can use pytest-runner to add setup.py test support for pytest
 runner.
 
+Deprecation Notice
+==================
+
+pytest-runner depends on deprecated features of setuptools and relies on features that break security
+mechanisms in pip. For example `setup_requires` and `tests_require` bypass `pip --require-hashes`.
```
Comment (graingert) [inline README.rst]:
```suggestion
mechanisms in pip. For example 'setup_requires' and 'tests_require' bypass ``pip --require-hashes``.
```

File: README.rst [hunk 4]
```diff
@@ -19,6 +19,21 @@
...
+pytest-runner depends on deprecated features of setuptools and relies on features that break security
+mechanisms in pip. For example 'setup_requires' and 'tests_require' bypass ``pip --require-hashes``.
+See also https://github.com/pypa/setuptools/issues/1684.
+
+It is recommended that you:
+
+- Remove 'pytest-runner' from your 'setup_requires', preferably removing the `setup_requires` option.
+- Remove 'pytest' and any other testing requirements from 'tests_require', preferably removing the `setup_requires` option.
+- Use a requirement-test.txt file to hold test dependencies
+- Invoke tests with ``pytest``
```
Comment (graingert) [inline README.rst]:
```suggestion
- Invoke tests with ``pip install --require-hashes requirements-test.txt && pytest``
```

File: README.rst [hunk 5]
```diff
@@ -19,6 +19,21 @@
...
+pytest-runner depends on deprecated features of setuptools and relies on features that break security
+mechanisms in pip. For example 'setup_requires' and 'tests_require' bypass ``pip --require-hashes``.
+See also https://github.com/pypa/setuptools/issues/1684.
+
+It is recommended that you:
+
+- Remove 'pytest-runner' from your 'setup_requires', preferably removing the `setup_requires` option.
+- Remove 'pytest' and any other testing requirements from 'tests_require', preferably removing the `setup_requires` option.
+- Use a requirement-test.txt file to hold test dependencies
+- Invoke tests with ``pip install --require-hashes requirements-test.txt && pytest``
```
Comment (jaraco) [inline README.rst]: This syntax is hostile to Windows and also mutates the user's environment. Especially for a user coming from `pytest-runner`, which doesn't mutate the environment, that might be a surprise.  If a one-liner is required, I'd suggest instead `pip-run -r tests/requirements.txt -- -m pytest`. Of course, that assumes pip-run and not just pip. I think better would be to recommend a generic "select a tool to bootstrap and then run tests such as tox". Maybe link to tox or link to a section in Python packaging docs that may enumerate options and/or give the (current) best recommendation.

File: README.rst (hunk 1 above)
Comment (jaraco) [reply README.rst]: As long as the package exists, I think having docs is worthwhile.

File: README.rst [hunk 6]
```diff
@@ -19,6 +19,21 @@
...
+
+pytest-runner depends on deprecated features of setuptools and relies on features that break security
+mechanisms in pip. For example 'setup_requires' and 'tests_require' bypass ``pip --require-hashes``.
+See also https://github.com/pypa/setuptools/issues/1684.
+
+It is recommended that you:
+
+- Remove 'pytest-runner' from your 'setup_requires', preferably removing the `setup_requires` option.
+- Remove 'pytest' and any other testing requirements from 'tests_require', preferably removing the `setup_requires` option.
+- Use a requirement-test.txt file to hold test dependencies
```
[the remark this answers is not in the document]
Comment (jaraco) [reply README.rst]: I don't think there is such a section yet. I wish there was.

**Pull request merged**